### DISCUSSION TITLE
SVGs being served as blank files

### DISCUSSION THREAD

**Discussion by NomadicDeveloper22** [pocketbase/pocketbase#2952] [Q&A] [answered]
I have a production server on fly.io running 0.16.10

If I upload a png through the admin console (the /_/ page), it uploads fine

If I upload a svg the same way, it seems to almost corrupt it. Downloading it from the same page shows its a blank file

If I download it from Cloudflare R2 (my S3 storage), it shows the svg perfectly fine

If I try to upload or view the svg from my flutter app, it doesn't through any error at any part of the process. Just doesn't load the svg. But it does upload it. However it is still essentially a blank file when downloaded through the web ui

Before uploading my pocketbase instance to fly.io, I ran it locally with no issues. However the difference is the local copy doesn't use R2/S3

I also used a 0.15.3 build on pockethose.io any that has no issues either, but again no R2/S3 here

If the svg is successfully uploaded to the R2/S3 bucket, and I can view it perfectly when downloading directly from it. How come pocketbase on fly.io can't?

I previously had 0.16.5 on fly.io until now, thinking the update would fix it. But the issue persists with 0.16.10

**Comment by NomadicDeveloper22** [comment]


**Reply by ganigeorgiev** [reply]
This seems similar to https://github.com/pocketbase/pocketbase/discussions/2068, but I don't have R2 account to test it and to inspect whether it modifies the uploaded svg somehow before storing it and what actually it returns for the getObject call.

**Reply by ganigeorgiev** [reply, marked as answer]
I wasn't able to reproduce the blank JSON/SVG issue with the S3 offering of Wasabi and Vultr.

It is most likely either some obscure R2 incompatibility with the S3 API OR a bug in the `aws-sdk-go` library (there is a very similar issue in https://github.com/aws/aws-sdk-go/issues/4697).

**Comment by devakrishna33** [comment]
I had the same issue, I have uploaded the SVG file using the wrangler cli it did not update the metadata correctly. When I uploaded through the R2 dashboard it seems to be fine.
